java/vector/src/main/codegen/templates/UnionListWriter.java

```diff
@@ -124,9 +145,15 @@ public ListWriter list() {
   }
 
   @Override
+  public ListWriter list(int size) {
+    // prime the writer with the fixed size list
+    writer.getWriter(new FixedSizeList(size));
```

Comment:
we don't really need to got Through ArrowType nor UnionType to call getWriter(...).
We could have a getFixedWidthList(size) in writer.